### ISSUE TITLE
AMD don't start

### ISSUE THREAD

**Issue by kk12l** [fireice-uk/xmr-stak#1721]
CPU Ryzen 1700
GPU VEGA56
```
-------------------------------------------------------------------
xmr-stak 2.4.6 64961dab

Brought to you by fireice_uk and psychocrypt under GPLv3.
Based on CPU mining code by wolf9466 (heavily optimized by fireice_uk).
Based on NVIDIA mining code by KlausT and psychocrypt.
Based on OpenCL mining code by wolf9466.

Configurable dev donation level is set to 2.0%

You can use following keys to display reports:
'h' - hashrate
'r' - results
'c' - connection
-------------------------------------------------------------------
[2018-07-17 09:57:20] : Mining coin: monero7
[2018-07-17 09:57:20] : Starting NVIDIA GPU thread 0, no affinity.
[2018-07-17 09:57:20] : Compiling code and initializing GPUs. This will take a while...
[2018-07-17 09:57:24] : Device 0 work size 8 / 32.
[2018-07-17 09:57:24] : OpenCL device 0 - Precompiled code C:\Users\Fabric\AppData\Local/.openclcache/b314032c6120f8815bea9cbf746b0e422de8456f1f7368be3985207395f283ed.openclbin not found. Compiling ...
[2018-07-17 09:57:24] : Error CL_BUILD_PROGRAM_FAILURE when calling clBuildProgram.
Build log:
"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 161: error: vector
          subscript not support
        k[0] ^= AES0[BYTE(x[0], 0)] ^ AES1[BYTE(x[1], 1)] ^ AES2[BYTE(x[2], 2)] ^ AES3[BYTE(x[3], 3)];
          ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 161: error: vector
          subscript not support
        k[0] ^= AES0[BYTE(x[0], 0)] ^ AES1[BYTE(x[1], 1)] ^ AES2[BYTE(x[2], 2)] ^ AES3[BYTE(x[3], 3)];
                     ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 161: error: vector
          subscript not support
        k[0] ^= AES0[BYTE(x[0], 0)] ^ AES1[BYTE(x[1], 1)] ^ AES2[BYTE(x[2], 2)] ^ AES3[BYTE(x[3], 3)];
                                           ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 161: error: vector
          subscript not support
        k[0] ^= AES0[BYTE(x[0], 0)] ^ AES1[BYTE(x[1], 1)] ^ AES2[BYTE(x[2], 2)] ^ AES3[BYTE(x[3], 3)];
                                                                 ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 161: error: vector
          subscript not support
        k[0] ^= AES0[BYTE(x[0], 0)] ^ AES1[BYTE(x[1], 1)] ^ AES2[BYTE(x[2], 2)] ^ AES3[BYTE(x[3], 3)];
                                                                                       ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 162: error: vector
          subscript not support
        x[0] ^= k[0];
          ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 162: error: vector
          subscript not support
        x[0] ^= k[0];
                  ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 163: error: vector
          subscript not support
        k[1] ^= AES0[BYTE(x[1], 0)] ^ AES1[BYTE(x[2], 1)] ^ AES2[BYTE(x[3], 2)] ^ AES3[BYTE(x[0], 3)];
          ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 163: error: vector
          subscript not support
        k[1] ^= AES0[BYTE(x[1], 0)] ^ AES1[BYTE(x[2], 1)] ^ AES2[BYTE(x[3], 2)] ^ AES3[BYTE(x[0], 3)];
                     ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 163: error: vector
          subscript not support
        k[1] ^= AES0[BYTE(x[1], 0)] ^ AES1[BYTE(x[2], 1)] ^ AES2[BYTE(x[3], 2)] ^ AES3[BYTE(x[0], 3)];
                                           ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 163: error: vector
          subscript not support
        k[1] ^= AES0[BYTE(x[1], 0)] ^ AES1[BYTE(x[2], 1)] ^ AES2[BYTE(x[3], 2)] ^ AES3[BYTE(x[0], 3)];
                                                                 ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 163: error: vector
          subscript not support
        k[1] ^= AES0[BYTE(x[1], 0)] ^ AES1[BYTE(x[2], 1)] ^ AES2[BYTE(x[3], 2)] ^ AES3[BYTE(x[0], 3)];
                                                                                       ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 164: error: vector
          subscript not support
        x[1] ^= k[1];
          ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 164: error: vector
          subscript not support
        x[1] ^= k[1];
                  ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 165: error: vector
          subscript not support
        k[2] ^= AES0[BYTE(x[2], 0)] ^ AES1[BYTE(x[3], 1)] ^ AES2[BYTE(x[0], 2)] ^ AES3[BYTE(x[1], 3)];
          ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 165: error: vector
          subscript not support
        k[2] ^= AES0[BYTE(x[2], 0)] ^ AES1[BYTE(x[3], 1)] ^ AES2[BYTE(x[0], 2)] ^ AES3[BYTE(x[1], 3)];
                     ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 165: error: vector
          subscript not support
        k[2] ^= AES0[BYTE(x[2], 0)] ^ AES1[BYTE(x[3], 1)] ^ AES2[BYTE(x[0], 2)] ^ AES3[BYTE(x[1], 3)];
                                           ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 165: error: vector
          subscript not support
        k[2] ^= AES0[BYTE(x[2], 0)] ^ AES1[BYTE(x[3], 1)] ^ AES2[BYTE(x[0], 2)] ^ AES3[BYTE(x[1], 3)];
                                                                 ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 165: error: vector
          subscript not support
        k[2] ^= AES0[BYTE(x[2], 0)] ^ AES1[BYTE(x[3], 1)] ^ AES2[BYTE(x[0], 2)] ^ AES3[BYTE(x[1], 3)];
                                                                                       ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 166: error: vector
          subscript not support
        x[2] ^= k[2];
          ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 166: error: vector
          subscript not support
        x[2] ^= k[2];
                  ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 167: error: vector
          subscript not support
        k[3] ^= AES0[BYTE(x[3], 0)] ^ AES1[BYTE(x[0], 1)] ^ AES2[BYTE(x[1], 2)] ^ AES3[BYTE(x[2], 3)];
          ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 167: error: vector
          subscript not support
        k[3] ^= AES0[BYTE(x[3], 0)] ^ AES1[BYTE(x[0], 1)] ^ AES2[BYTE(x[1], 2)] ^ AES3[BYTE(x[2], 3)];
                     ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 167: error: vector
          subscript not support
        k[3] ^= AES0[BYTE(x[3], 0)] ^ AES1[BYTE(x[0], 1)] ^ AES2[BYTE(x[1], 2)] ^ AES3[BYTE(x[2], 3)];
                                           ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 167: error: vector
          subscript not support
        k[3] ^= AES0[BYTE(x[3], 0)] ^ AES1[BYTE(x[0], 1)] ^ AES2[BYTE(x[1], 2)] ^ AES3[BYTE(x[2], 3)];
                                                                 ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 167: error: vector
          subscript not support
        k[3] ^= AES0[BYTE(x[3], 0)] ^ AES1[BYTE(x[0], 1)] ^ AES2[BYTE(x[1], 2)] ^ AES3[BYTE(x[2], 3)];
                                                                                       ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 209: warning: variable
          "SKEIN256_IV" was declared but never referenced
  static const __constant ulong SKEIN256_IV[8] =
                                ^

"C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl", line 676: warning: variable
          "c_Padding" was declared but never referenced
  __constant static const sph_u32  c_Padding[16] = {
                                   ^

26 errors detected in the compilation of "C:\Users\Fabric\AppData\Local\Temp\OCL8816T1.cl".
Frontend phase failed compilation.

[2018-07-17 09:57:25] : WARNING: AMD device not found
[2018-07-17 09:57:25] : WARNING: backend AMD (OpenCL) disabled.
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 0.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 2.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 4.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 6.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 8.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 10.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 12.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Starting 1x thread, affinity: 14.
[2018-07-17 09:57:25] : hwloc: memory pinned
[2018-07-17 09:57:25] : Fast-connecting to xmr-eu2.nanopool.org:14444 pool ...
[2018-07-17 09:57:25] : Pool xmr-eu2.nanopool.org:14444 connected. Logging in...
[2018-07-17 09:57:25] : Difficulty changed. Now: 120001.
[2018-07-17 09:57:25] : Pool logged in.
```

**Comment by psychocrypt** [comment]
This issue will be fixed with #1722 I try to create a bug fix release today.

**Comment by psychocrypt** [comment]
fixed in the release 2.4.7 https://github.com/fireice-uk/xmr-stak/releases
